libcst/metadata/scope_provider.py

```diff
@@ -851,33 +849,37 @@ def visit_Attribute(self, node: cst.Attribute) -> Optional[bool]:
 
     def visit_Call(self, node: cst.Call) -> Optional[bool]:
         self.__top_level_attribute_stack.append(None)
+        self.__in_type_hint_stack.append(False)
+        self.__in_annotation_stack.append(False)
         qnames = {qn.name for qn in self.scope.get_qualified_names_for(node)}
         if "typing.NewType" in qnames or "typing.TypeVar" in qnames:
             node.func.visit(self)
-            self.__in_type_hint.add(node)
+            self.__in_type_hint_stack.pop()
+            self.__in_type_hint_stack.append(True)
```

Comment:
This is almost 2x faster
```suggestion
            self.__in_type_hint_stack[-1] = True
```